src/sentry/snuba/metrics/extraction.py

```diff
@@ -354,7 +405,7 @@ def query_hash(self) -> str:
         """Returns a hash of the query and field to be used as a unique identifier for the on-demand metric."""
 
         # TODO: Figure out how to support multiple fields and different but equivalent queries
-        str_to_hash = f"{self.field};{self._query}"
+        str_to_hash = f"{self.field};{self._field_condition};{self._query}"
```

Comment:
@jan-auer this is changing the query hashing algorithm, we need to understand how to move forward here. I would roll it out but also add the deep sorting to that we can already use the AST.

The deep sorting would look like:
```
def _deep_sorted(value: Union[Any, Dict[Any, Any]]) -> Union[Any, Dict[Any, Any]]:
    if isinstance(value, dict):
        return {key: _deep_sorted(value) for key, value in sorted(value.items())}
    else:
        return value
```

Reply:
Done in [6fd5485](https://github.com/getsentry/sentry/pull/54092/commits/6fd5485c1d7adc59621366eed15e2665243697c3)